openpype/pipeline/temporarydir.py

```diff
@@ -0,0 +1,65 @@
+"""
+Temporary folder operations
+"""
+
+import os
+from openpype.lib import StringTemplate
+from openpype.pipeline import Anatomy
+
+
+def create_custom_tempdir(project_name, anatomy=None, formating_data=None):
+    """ Create custom tempdir
+
+    Template path formatting is supporting:
+    - optional key formating
+    - available keys:
+        - root[work | <root name key>]
+        - project[name | code]
+
+    Args:
+        project_name (str): name of project
+        anatomy (openpype.pipeline.Anatomy): Anatomy object
+        formating_data (dict): formating data used for filling template.
+
+    Returns:
+        bool | str: formated path or None
+    """
+    openpype_tempdir = os.getenv("OPENPYPE_TMPDIR")
+    if not openpype_tempdir:
+        return
+
+    custom_tempdir = None
+    if "{" in openpype_tempdir:
+        if anatomy is None:
+            anatomy = Anatomy(project_name)
+        # create base formate data
+        data = {
+            "root": anatomy.roots
+        }
+        if formating_data is None:
+            # We still don't have `project_code` on Anatomy...
+            project_doc = anatomy.get_project_doc_from_cache(project_name)
+            data["project"] = {
+                "name": project_name,
+                "code": project_doc["data"]["code"],
+            }
```

Comment:
After this [PR](https://github.com/ynput/OpenPype/pull/4445) can be modified to
```suggestion
            data["project"] = {
                "name": project_name,
                "code": anatomy.project_code,
            }
```